From gadiyar@ti.com  Wed Apr 28 12:36:54 2010
From: Anand Gadiyar <gadiyar@ti.com>
Date: Mon, 29 Mar 2010 16:54:51 +0530
Subject: USB: fix build on OMAPs if CONFIG_PM_RUNTIME is not set
To: linux-usb@vger.kernel.org, Greg KH <gregkh@suse.de>
Cc: Anand Gadiyar <gadiyar@ti.com>, Michal Marek <mmarek@suse.cz>, Tony Lindgren <tony@atomide.com>, Alan Stern <stern@rowland.harvard.edu>
Message-ID: <1269861891-25841-1-git-send-email-gadiyar@ti.com>


With patch as1329 (USB: convert to the runtime PM framework),
we make USB_SUSPEND depend on PM_RUNTIME instead of CONFIG_PM.

Also, CONFIG_USB_OTG selects CONFIG_USB_SUSPEND.

If PM_RUNTIME is not enabled, and we try to enable USB_OTG,
we will end up with CONFIG_USB_SUSPEND selected. This is
due to a known bug with the select statement.

This makes the build break on various OMAP configs (which
have CONFIG_USB_OTG set by default, but do not yet have
CONFIG_PM_RUNTIME enabled).

Avoid this by changing the logic for CONFIG_USB_OTG from
"select USB_SUSPEND" to "depends on USB_SUSPEND"

Signed-off-by: Anand Gadiyar <gadiyar@ti.com>
CC: Michal Marek <mmarek@suse.cz>
CC: Tony Lindgren <tony@atomide.com>
CC: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/Kconfig
+++ b/drivers/usb/core/Kconfig
@@ -109,7 +109,7 @@ config USB_SUSPEND
 config USB_OTG
 	bool
 	depends on USB && EXPERIMENTAL
-	select USB_SUSPEND
+	depends on USB_SUSPEND
 	default n
 
 
